nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -20,16 +18,149 @@ Arrows indicate that the elements can communicate with each other directly.
 
    Modem library OS abstraction overview
 
-Creating the OS abstraction layer
-*********************************
+To use the modem library, certain requirements need to be fulfilled:
+
+* The modem library makes use of the IPC driver in the nrfx library.
+* For communication with the modem a shared memory area in the lower 128kB of RAM is required.
+* The application needs to be running in the non-secure domain.
+* The OS abstraction :file:`nrf_modem_os.c` needs to be implemented.
+
+Obtaining the Modem library
+***************************
+
+The application can either include nrfxlib entirely or just the modem library.
+Header files are located in :file:`nrf_modem/include/`. The library binary can be found in :file:`nrf_modem/lib/`, both for hard-float and soft-float.
+
+IPC Driver and Interrupt
+************************
+
+The Modem library uses the IPC peripheral for communication with the modem and makes use of the nrfx IPC driver for that purpose.
+The application must either include the nrfx IPC driver or implement its own IPC driver with the same signature as the nrfx IPC driver.
+
+If the OS has its own IRQ handler scheme that does not directly forward the IPC_IRQHandler, the OS must route the IPC_IRQHandler to the nrfx IPC IRQ handler.
+
+Application Interrupt
+*********************
+The Modem library receives events from the modem on the high-priority IPC interrupt.
+To continue processing these events on a lower priority, an application interrupt is used.
+The library calls the :c:func:`nrf_modem_os_application_irq_set` function to trigger the interrupt.
+The handler for the low-priority interrupt must then call the :c:func:`nrf_modem_application_irq_handler` function and subsequently wake up all sleeping threads using the :c:func:`nrf_modem_os_event_notify` function.
+
+The abstraction layer decides the implementation of the application interrupt.
+Possible solutions for this are either directly setting an IRQ or using an EGU event.
+
+Memory
+******
+
+The Modem library requires two types of memory: A library heap and shared memory with the modem.
+The OS abstraction implementation is responsible for providing a dynamic memory allocation for both cases.
+
+Library Heap
+============
+
+The library heap is used by the modem for dynamic memory allocations.
+It has no location requirements and may use the OS provided heap implementation.
+
+Shared Memory
+=============
+
+The shared memory regions have a fixed size and need to reside in the lower 128kB of RAM.
```

Comment:
From the Modem library's perspective, these are configurable at runtime via the initialization parameters.